### PR TITLE
🐛 Fix "can't find referenced class" caused by desugaring

### PR DESCRIPTION
**Pull request by hammerhai**
[REFERENCE 602](https://community.kodular.io/t/rush-a-new-and-improved-way-of-building-extensions/111470/602) within topic.

```java
// After `min_sdk` set to 24 or higher
import android.app.Application.ActivityLifecycleCallbacks;
```

```java
// Before `min_sdk` set to 24 or higher
import android.app.Application.ActivityLifecycleCallbacks;
import android.app.Application.ActivityLifecycleCallbacks..CC;
```

### CODE REVIEW

File: lib/src/commands/build/tools/executor.dart [hunk 1]
```diff
@@ -15,10 +15,11 @@ class Executor {
   static final _libService = GetIt.I<LibService>();
   static final _processRunner = ProcessRunner();
 
-  static Future<void> execD8(String artJarPath) async {
+  static Future<void> execD8(Config config, String artJarPath) async {
     final args = <String>[
       ...['-cp', await _libService.r8Jar()],
       'com.android.tools.r8.D8',
+      ...['--min-api', '${config.minSdk}'],
```
Comment (shreyashsaitwal) [inline lib/src/commands/build/tools/executor.dart:22]: IDK if I remember it correctly but APK generation fails if an extension's DEX file is compiled with a higher min-api than that of AI2. Correct me if I'm wrong or if this has changed recently.

**Review by shreyashsaitwal — CHANGES_REQUESTED** [review]

File: lib/src/commands/build/tools/executor.dart (hunk 1 above)
Comment (hammerhai) [reply lib/src/commands/build/tools/executor.dart:22]: I have compiled successfully using App Inventor, with a `min_sdk` of 24!

**Review by shreyashsaitwal — COMMENTED** [review]
Lgtm, thanks!

**Review by shreyashsaitwal — APPROVED** [review]
Lgtm, thanks!

**Pull request merged**